### ISSUE TITLE
Add Log Configuration to trace what is injected in the App

### ISSUE THREAD

**Issue by davidrevoledo** [davidrevoledo/Inyector#5]
As Dmitry Pavlov has suggested, it'd be useful trace what is being injected in the App.

I was thinking in a configuration like 

`c.EnableTracing = true`

the problem with this is that it doesn't make really sense to trace that on PRD and a configuration require a flag check everytime the registration is being processed.

Other possibility could be just in not Release Mode just Log anyway  in the Output and Give the possibility to over write that similar how is doing EF with the sql generated.

DataBase.Log = sql = > {

});

That way we can have a Action<Type,Type> to do whatever we can with tracing.

**Comment by davidrevoledo** [comment]
https://github.com/davidrevoledo/Inyector/pull/6

**Comment by davidrevoledo** [comment]
Will be merged in v 1.3 with others requested features

**Comment by dmitry-pavlov** [comment]
As for logging, I guess it is always makes sense to use standard things. I would use [ILogger](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.ilogger) interface internally. 

As for configurring it, this can be something like  **ConfigureLogging** extention for your `configurations` chain.

```
 configurations.Scan(typeof(Startup).Assembly)
                    .DefaultMode(services, ServiceLifetime.Singleton)
                    .AddRuleForNamingConvention(ServiceLifetime.Singleton)
                    .ConfigureLogging(options => 
                   {
                         options.LogLevel = LogLevel.Trace;
                         options. ... // any other configuration
                   });
```
[LogLevel](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.logging.loglevel) is just one logging options for example.  **ConfigureLogging**

**Comment by davidrevoledo** [comment]
@dmitry-pavlov yeah that really make sense I'm changing a few stuffs in other branch to support DI internally as I need to create some tests.

**Comment by davidrevoledo** [comment]
@dmitry-pavlov just to know what other types of tracing do you think the library can support ? thanks
